tests/integration/conftest.py

```diff
@@ -185,16 +227,73 @@ async def machine_model_fixture(
 
 
 @pytest_asyncio.fixture(scope="function", name="jenkins_machine_agent")
-async def jenkins_machine_agent_fixture(machine_model: Model) -> Application:
+async def jenkins_machine_agent_fixture(
+    machine_model: Model, app_suffix: str
+) -> typing.AsyncGenerator[Application, None]:
     """The jenkins machine agent."""
     # 2023-06-02 use the edge version of jenkins agent until the changes have been promoted to
     # stable.
     app = await machine_model.deploy(
-        "jenkins-agent", channel="latest/edge", config={"labels": "machine"}
+        "jenkins-agent",
+        channel="latest/edge",
+        config={"labels": "machine"},
+        application_name=f"jenkins-agent-{app_suffix}",
     )
+    await machine_model.create_offer(f"{app.name}:slave")
     await machine_model.wait_for_idle(apps=[app.name], status="blocked", timeout=1200)
 
-    return app
+    yield app
+
+    await machine_model.remove_offer(f"admin/{machine_model.name}.{app.name}", force=True)
+    await machine_model.remove_application(app.name, force=True)
+
+
+@pytest_asyncio.fixture(scope="function", name="jenkins_multi_machine_agents")
+async def jenkins_multi_machine_agents_fixture(
+    machine_model: Model, num_units: int, app_suffix: str
+) -> typing.AsyncGenerator[Application, None]:
+    """The jenkins machine agent with 3 units."""
+    # 2023-06-02 use the edge version of jenkins agent until the changes have been promoted to
+    # stable.
+    app = await machine_model.deploy(
+        "jenkins-agent",
+        channel="latest/edge",
+        config={"labels": "machine"},
+        application_name=f"jenkins-agent-{app_suffix}",
+        num_units=num_units,
```

Comment:
Its the same fixture as the previous one but with num_units specified, is there a way to parametrize this ?

Reply:
The multi-unit fixture is used for the new relation, multi-units for old relation will fail, hence the differentiation without parametrization. 
Shall I rename this fixture to perhaps something like `new_relation_agent`?

Reply:
Changed! Thank you for the review!